task/04-arrays-tasks.js

```diff
@@ -529,37 +602,49 @@ function selectMany(arr, childrenSelector) {
  * @param {array} arr
  * @param {array} indexes
  * @return {any} element from array
- * 
+ *
  * @example
- *   [[1, 2], [3, 4], [5, 6]], [0,0]  => 1        (arr[0][0]) 
- *   ['one','two','three'], [2]       => 'three'  (arr[2]) 
+ *   [[1, 2], [3, 4], [5, 6]], [0,0]  => 1        (arr[0][0])
+ *   ['one','two','three'], [2]       => 'three'  (arr[2])
  *   [[[ 1, 2, 3]]], [ 0, 0, 1 ]      => 2        (arr[0][0][1])
  */
 function getElementByIndexes(arr, indexes) {
-    throw new Error('Not implemented');
+    let newArr = [...arr]
+    indexes.map(el => {
+        newArr = newArr[el]
+    })
+    return newArr
 }
 
 
 /**
  * Swaps the head and tail of the specified array:
- * the head (first half) of array move to the end, the tail (last half) move to the start. 
+ * the head (first half) of array move to the end, the tail (last half) move to the start.
  * The middle element (if exists) leave on the same position.
- * 
- *  
+ *
+ *
  * @param {array} arr
  * @return {array}
- * 
+ *
  * @example
  *   [ 1, 2, 3, 4, 5 ]   =>  [ 4, 5, 3, 1, 2 ]
- *    \----/   \----/         
- *     head     tail 
+ *    \----/   \----/
+ *     head     tail
+ *
+ *   [ 1, 2 ]  => [ 2, 1 ]
+ *   [ 1, 2, 3, 4, 5, 6, 7, 8 ]   =>  [ 5, 6, 7, 8, 1, 2, 3, 4 ]
  *
- *   [ 1, 2 ]  => [ 2, 1 ] 
- *   [ 1, 2, 3, 4, 5, 6, 7, 8 ]   =>  [ 5, 6, 7, 8, 1, 2, 3, 4 ]   
- * 
  */
 function swapHeadAndTail(arr) {
-    throw new Error('Not implemented');
+    let headArr = arr.filter((el,i)=>{
+        if (i<Math.floor(arr.length/2)) return true
+    })
+    let tailArr = arr.filter((el,i)=>{
+        if (i>=Math.ceil(arr.length/2)) return true
+    })
+    let middle = arr.length%2===0 ? [] : [arr[Math.floor(arr.length/2)]]
+
+    return arr.length===2 ? [arr[1],arr[0]] : [tailArr.concat(middle.concat(headArr))].flat()
```

Comment:
spaces around operators

Please, avoid this code style, if-else is more readable than multiple conditional operators ? :